lib/matplotlib/sphinxext/plot_directive.py

```diff
@@ -494,10 +486,7 @@ def run_code(code, code_path, ns=None, function_name=None):
 
     # Redirect stdout
     stdout = sys.stdout
-    if six.PY3:
-        sys.stdout = io.StringIO()
-    else:
-        sys.stdout = cStringIO.StringIO()
+    sys.stdout = io.StringIO()
```

Comment:
see if contextlib.redirect_stdout works here

Reply:
*probably* would, so I'm leaving it for later.